File: examfront/src/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor( private http : HttpClient) { }

  getAllCategory(){
    return this.http.get(`${baseUrl}/category/all-category`);
  }

  getSingleCategory(cid : any){
    return this.http.get(`${baseUrl}/category/${cid}`)
  }
  addCategory(category : any)
  {
    return this.http.post(`${baseUrl}/category/add-category`,category);
  }
  deleteCategory(categoryId : any)
  {
   // console.log(`${baseUrl}/category/152`);
    return this.http.delete(`${baseUrl}/category/${categoryId}`);
  }
  updateCategory(category : any)
  {
    console.log("put mapping");
    return this.http.put(`${baseUrl}/category/update-category/`,category);
  }
}

File: JwtExample3/JwtExample3/src/main/java/com/jwt/example/service/CategoryService.java
package com.jwt.example.service;

import java.util.Set;

import com.jwt.example.entity.exam.Category;

public interface CategoryService {
	public Category addCategory(Category c);
	public Category updateCategory(Category c);
	public Set<Category> getCategories();
	public Category getCategory(Long cId);
	public void deleteCategory(Long cId);

}
